### ISSUE TITLE
array.pop() not working

### ISSUE THREAD

**Issue by pierznj** [robertkrimen/otto#225]
Some code we have which used <some string with forward slashes>.split('/').pop() no longer works, instead throwing a TypeError.

**Comment by deoxxa** [comment]
More details please :D

**Comment by pierznj** [comment]
Apologies - I used angle braces and they got cut. More or less we have:
```                   
var job = {
    recipient: 'Practitioner/some_id'
};
var recipientId=job.recipient.split('/').pop();
```
This throws a TypeError. The following, however, does not:

```
var job = {
    recipient: 'Practitioner/some_id'
};
var recipientId=job.recipient.split('/')[1];
```

**Comment by deoxxa** [comment]
```
> var job = { recipient: 'Practitioner/some_id' }
undefined
> var recipientId=job.recipient.split('/').pop();
undefined
> recipientId
"some_id"
```

Can't seem to reproduce it - do you have a stack trace?

**Comment by pierznj** [comment]
No stack trace is provided, least not that I can find. I looked for error.prototype.stack. The error message is a terse "TypeError"; that's it.  The code provided is not what we actually have, but it's the essence of it:
```
                        ...
			var cq = {
				'recipient:Practitioner': job.recipient.split('/').pop(),
				category: 'PUSH',
			};
			var comms = tx.search('Communication', cq).entry;
                        ...
```
job. recipient == 'Practitioner/some_id' at this point.

**Comment by dhiltonp** [comment]
This is caused by my pr #218, which made string splits much faster.

There was a bunch of overhead building an array that didn't quite make sense, as a single call built an array much faster. It turns out that the old code was building value of type "Array", while the newer code returns a type of "GoArray".

I looked into it and found no way of converting an Array to a GoArray faster than the prior code. Adding javascript methods to GoArray would help, but it would never be exactly the same due to type restrictions.

I've added a pr that will revert this change :'(
